ui/store/actions.js

```diff
@@ -1116,6 +1126,37 @@ export function updateMetamaskState(newState) {
       type: actionConstants.UPDATE_METAMASK_STATE,
       value: newState,
     });
+
+    // Check that the transaction was not submitted successfully, and remove it from failed transactions if it was.
+    const transactionIdsToRemove = Object.keys(
+      transactionsToDisplayOnFailure,
+    ).filter((id) => {
+      const foundTx = currentNetworkTxList.find((currentTx, index) => {
```

Comment:
I don't think this is doing what you meant it to do. It looks like this is finding the first transaction in `currentNetworkTxList` that has changed to a relevant status, irrespective of the id. Then it's checking the ID last. This will find at-most a single transaction.